pkg/redis/redis.go

```diff
@@ -76,12 +75,38 @@ type Client struct {
 
 // Config represents Redis configuration.
 type Config struct {
-	config.Redis
-	Namespace []string
+	Address       string         `name:"address" description:"Address of the Redis server"`
+	Password      string         `name:"password" description:"Password of the Redis server"`
+	Database      int            `name:"database" description:"Redis database to use"`
+	RootNamespace []string       `name:"namespace" description:"Namespace for Redis keys"`
+	PoolSize      int            `name:"pool-size" description:"The maximum number of database connections"`
+	Failover      FailoverConfig `name:"failover" description:"Redis failover configuration"`
+	namespace     []string
+}
+
+func (c Config) WithNamespace(namespace ...string) *Config {
+	deriv := c
+	deriv.namespace = namespace
```

Comment:
This should be `append` IMO. Otherwise this method should be called `SetNamespace`.

Reply:
I don't think so. That might result in accidentally concatenating namespaces.

Namespace is one value. That is also why it is singular `namespace` and not `namespaces`. It is passed as a string slice but that is to leave the delimiter up to the store.